src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,190 @@
+import math
+import rospy
+import tf2_ros
+import tf2_geometry_msgs
+import threading
+
+from geometry_msgs.msg import TwistStamped
+from geometry_msgs.msg import Point
+from geometry_msgs.msg import PointStamped
+
+from task_utilities.acceleration_limiter import AccelerationLimiter
+
+from .abstract_task import AbstractTask
+from iarc_tasks.task_states import (TaskRunning,
+                                    TaskDone,
+                                    TaskCanceled,
+                                    TaskAborted,
+                                    TaskFailed)
+from iarc_tasks.task_commands import (VelocityCommand, NopCommand)
+
+from translate_stop_planner import TranslateStopPlanner
+
+class GoToRoombaState(object):
+    init = 0
+    translate = 1
+
+class GoToRoombaTask(AbstractTask):
+
+    def __init__(self, task_request):
+        super(GoToRoombaTask, self).__init__()
+
+        self._roomba_id = task_request.frame_id + '/base_link'
+
+        if self._roomba_id is None:
+            raise ValueError('A null roomba id was provided')
+
+        self._roomba_odometry = None
+        self._roomba_point = None
+        self._limiter = AccelerationLimiter()
+
+        self._canceled = False;
+        self._last_update_time = None
+        self._current_velocity = None
+        self._transition = None
+
+        self._lock = threading.RLock()
+
+        self._x_position = task_request.x_position
+        self._y_position = task_request.y_position
+
+
+        try:
+            self._TRANSLATION_XYZ_TOLERANCE = rospy.get_param('~translation_xyz_tolerance')
+            self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._MIN_MANEUVER_HEIGHT = rospy.get_param('~min_maneuver_height')
+            self._MAX_HORIZ_SPEED = rospy.get_param('~max_translation_speed')
+            self._MAX_TASK_DIST = rospy.get_param('~max_roomba_dist')
+            self._K_X = rospy.get_param('~k_term_tracking_x')
+            self._K_Y = rospy.get_param('~k_term_tracking_y')
+            self._z_position = rospy.get_param('~track_roomba_height')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for xyztranslation task')
+            raise
+
+        # Check that we aren't being requested to go below the minimum maneuver height
+        # Error straight out if that's the case. If we are currently below the minimum height
+        # It will be caught and handled on the next update
+        if self._z_position < self._MIN_MANEUVER_HEIGHT :
+            raise ValueError('Requested z height was below the minimum maneuver height')
+
+        # fc status?
```

Comment:
What does this comment mean?

Reply:
I added it a while ago, I don't remember what it does, so I remove it